### PR TITLE
Adds support for multiple bounding box formats (xywh and cxcywh) to the mean_iou_bbox function, addressing the limitation mentioned in issue #3529.

### LINKED ISSUE #3529: [Feature]: Support multiple bounding box formats in mean_iou_bbox (e.g., XYWH)
**Title:** Support multiple bounding box formats in `mean_iou_bbox` (e.g., XYWH)

### Description

The `mean_iou_bbox` function currently assumes that all input bounding boxes are strictly in the `(x1, y1, x2, y2)` (XYXY) coordinate format.

```python
# TODO: support more box types. e.g. xywh,
if not (((boxes_1[:, 2] - boxes_1[:, 0]) > 0).all() ...

```

This limitation restricts the function's usability with datasets or models that output boxes in other common formats, such as:

* `xywh` (top-left x, top-left y, width, height)
* `cxcywh` (center x, center y, width, height)

Currently, a user must manually convert their tensors before calling this function.

### Goal

Refactor `mean_iou_bbox` to accept an argument specifying the input box format and handle the conversion internally, or automatically detect/normalize the format.

### PR DESCRIPTION
**Pull request by arjitrawat15**
Description - Adds support for multiple bounding box formats (xywh and cxcywh) to the mean_iou_bbox function, addressing the limitation mentioned in issue #3529.

Changes -
1. Added _convert_boxes_to_xyxy() helper function to convert between box formats
2. Modified mean_iou_bbox() to accept a box_format parameter (default: xyxy)
3. Added comprehensive tests for all three supported formats

Supported Formats - 
1. xyxy : (x1, y1, x2, y2) - top-left and bottom-right corners (original format)
2. xywh : (x, y, w, h) - top-left corner with width and height
3. cxcywh : (cx, cy, w, h) - center point with width and height

Testing -
All tests pass (11/11):
- Original mean_iou tests (5/5)
- New bbox format tests (6/6)
  - XYXY format validation
  - XYWH format conversion
  - CXCYWH format conversion
  - Format consistency across all three formats
  - Invalid format error handling
  - Default format backward compatibility
 
image

Backward Compatibility - The default box_format=xyxy ensures complete backward compatibility with existing code.

Fixes #3529

### CODE REVIEW

**Comment by arjitrawat15** [comment]
Note on CI Failures - The 2 failing checks (collector and tutorials / tutorials (3.12)) are pre-existing issues in the main repository and are unrelated to this PR. 
This PR only modifies kornia/metrics/mean_iou.py and tests/metrics/test_mean_iou.py
All 33 other checks pass successfully, and all tests (11/11) pass locally.

File: kornia/metrics/mean_iou.py [hunk 1]
```diff
@@ -75,44 +75,98 @@ def mean_iou(pred: torch.Tensor, target: torch.Tensor, num_classes: int, eps: fl
...
+        y2 = y + h
+        return torch.stack([x, y, x2, y2], dim=1)
+    elif box_format == "cxcywh":
+        # (cx, cy, w, h) -> (x1, y1, x2, y2)
+        cx, cy, w, h = boxes[:, 0], boxes[:, 1], boxes[:, 2], boxes[:, 3]
+        x1 = cx - w / 2
+        y1 = cy - h / 2
+        x2 = cx + w / 2
+        y2 = cy + h / 2
+        return torch.stack([x1, y1, x2, y2], dim=1)
```
Comment (sidd-27) [inline kornia/metrics/mean_iou.py]: benchmark this against using torch.cat if possible

Comment (arjitrawat15) [reply kornia/metrics/mean_iou.py]: okay , i'll benchmark both and update soon.

**Comment by arjitrawat15** [comment]
Hey @sidd-27, Benchmarked torch.stack vs torch.cat on 10,000 boxes, 1,000 iterations. 
Results:
1. XYWH format: torch.cat is 1.09x faster (0.0639s vs 0.0588s)
2. CXCYWH format: torch.cat is 1.03x faster (0.1299s vs 0.1263s)
3. Average speedup:1.06x
image
Updated implementation to use torch.cat for better performance. All tests still pass (11/11).
image
